validator/validator_test.go

```diff
@@ -0,0 +1,45 @@
+package validator
+
+import (
+	"regexp"
+	"testing"
+)
+
+func TestProjectKey(t *testing.T) {
+	validProjectKeys := []string{
+		"abc", // min 3
+		"abcde12345", // max 10
+		"abc-123", // hyphen is supported but not documented
+	}
+
+	for _, projectKey := range validProjectKeys {
+		t.Run(projectKey, func(t *testing.T) {
+			diag := ProjectKey(projectKey, nil)
+			if diag.HasError() {
+				t.Errorf("ProjectKey validation failed. diag: %v", diag)
+			}
+		})
+	}
+}
+
+func TestProjectKey_invalidKeys(t *testing.T) {
+	invalidProjectKeys := []string{
+		"ab", // 2 characters, too short
+		"abcdefghijk", // 11 characters, too long
+		"abc,", // invalid characters
+	}
```

Comment:
I am glad we have some test cases. Can we also add `abcABdeF` and `AB123abc` so that we have mix of lower and upper case characters...